hadoop-pipeline-application/src/main/java/org/molgenis/hadoop/pipeline/application/partitioners/RegionSamRecordGroupingComparator.java

```diff
@@ -0,0 +1,31 @@
+package org.molgenis.hadoop.pipeline.application.partitioners;
+
+import org.apache.hadoop.io.WritableComparable;
+import org.apache.hadoop.io.WritableComparator;
+import org.molgenis.hadoop.pipeline.application.writables.RegionWithSortableSamRecordWritable;
+
+/**
+ * Custom grouping comparator for the {@link RegionWithSortableSamRecordWritable}, where only
+ * {@link RegionWithSortableSamRecordWritable#getRegionWritable()} is used within the grouping comparator (so the natural key
+ * part from the composite key). This comparator controls which keys are grouped together into a single call to the
+ * reduce() method.
+ */
+public class RegionSamRecordGroupingComparator extends WritableComparator
+{
+	public RegionSamRecordGroupingComparator()
+	{
+		// Gives WritableComparable class and the creation of instances for the keys is set to true.
+		super(RegionWithSortableSamRecordWritable.class, true);
+	}
+
+	@SuppressWarnings("rawtypes")
+	@Override
+	public int compare(WritableComparable first, WritableComparable second)
+	{
+		RegionWithSortableSamRecordWritable castedFirst = (RegionWithSortableSamRecordWritable) first;
+		RegionWithSortableSamRecordWritable castedSecond = (RegionWithSortableSamRecordWritable) second;
+
+		// Comparison is done only on the RegionWritable (natural key).
+		return castedFirst.getRegionWritable().compareTo(castedSecond.getRegionWritable());
```

Comment:
To quote [Hadoop's WritableComparator Javadoc](https://hadoop.apache.org/docs/r2.6.0/api/org/apache/hadoop/io/WritableComparator.html) (of which this class is a subclass from): "To define alternate orderings, override [compare(WritableComparable,WritableComparable)](https://hadoop.apache.org/docs/r2.6.0/api/org/apache/hadoop/io/WritableComparator.html#compare%28org.apache.hadoop.io.WritableComparable,%20org.apache.hadoop.io.WritableComparable%29)."